new addon purchase_requisition_auto_rfq

gurneyalex opened the pull request:
This addon adds a button to generate RFQ for the purchase requisition
using the product.supplierinfo in order to only have the lines corresponding to
the products sold by each supplier.

gurneyalex committed:
new addon purchase_requisition_auto_rfq

This addon adds a button to generate RFQ for the purchase requisition
using the product.supplierinfo in order to only have the lines corresponding to
the products sold by each supplier.

pedrobaeza commented on purchase_requisition_auto_rfq/model/purchase_requisition.py:
```diff
@@ -0,0 +1,71 @@
+                         'generated RFQs:<ul>%s</ul></p>')
+                body %= ''.join(u'<li>%s</li>' % product.name
+                                for product in products_without_supplier)
+                print body
```
Remove print.

gurneyalex commented:
@pedrobaeza good catch! Fixed.

pedrobaeza commented:
We have developed the same module at OdooMRP project (https://github.com/odoomrp/odoomrp-wip/tree/8.0/purchase_requisition_order_generator). What a pity!

I was going to make the PR next week, but now, I will compare both and propose to you possible enhancements (if any).

gurneyalex commented:
needs fixing: 
- [x] flake8
- [x] button does not show up as expected

gurneyalex commented:
@pedrobaeza sorry to here that. We can probably merge things; Fortunately, the module is quite small...

lepistone committed:
rephrase comments on tests

lepistone commented:
thanks @gurneyalex. I improved slightly some comments on the tests. :+1:

eLBati commented on purchase_requisition_auto_rfq/test/purchase_requisition.yml:
```diff
@@ -0,0 +1,43 @@
+    expected = {ref('base.res_partner_3'): [ref('kitchenset'), ref('blankets')],
+                ref('base.res_partner_2'): [ref('kitchenset')],
+                ref('base.res_partner_4'): [ref('blankets')],
+                }
```
Could you also check the quantities?

jgrandguillaume commented:
Hi,

Functional test made: Ok for me !

:+1: 

Regards,

Joël

joaoalf commented:
:+1:
lgtm

lepistone commented:
Hi, I'm improving the test to add the check suggested by @eLBati, then I'll merge.

Thanks!

lepistone committed:
refactor tests to be more readable

Here I take advantage of the new API and the assertions from nose.tools
to refactor the tests and make them more readable. They are still green
and should be otherwise unchanged.

lepistone committed:
make method called by button return True

The method auto_rfq_from_suppliers returned a list of ids of created
RFQs. This list was used by the tests only. In production a method
called by a button should just return True in case of success.

Now the method returns True, and tests are adapted to figure out the
created rfqs from requisition.purchase_ids instead of using the return
value.

lepistone committed:
improve assertion: compare dict with dict

This way assert_equal gives a very nice failure singling out the
different keys and values in the dictionary.

This is to prepare a test on product quantities, still getting good
failures if for example only the quantities for one line are different.

lepistone committed:
improve tests to check quantities as well

lepistone commented:
In the last commits I improved the tests as suggested by @elbati. I took the occasion to refactor them a bit and to remove a return value in production code that was only used in the tests.

eLBati commented:
:+1: :grin:
